traits/tests/test_configure_traits.py

```diff
@@ -56,6 +56,23 @@ def test_filename_but_no_file(self):
         self.assertIsInstance(unpickled, Model)
         self.assertEqual(unpickled.count, model.count)
 
+    def test_pickle_protocol(self):
+        # Pickled files should use protocol 3 (a compromise between
+        # efficiency and wide applicability).
+
+        model = Model(count=37)
+        filename = os.path.join(self.tmpdir, "nonexistent.pkl")
+        self.assertFalse(os.path.exists(filename))
+
+        with mock.patch.object(self.toolkit, "view_application"):
+            model.configure_traits(filename=filename)
+
+        self.assertTrue(os.path.exists(filename))
+        with open(filename, "rb") as pickled_object:
+            protocol = pickled_object.read(2)
+
+        self.assertEqual(protocol, b'\x80\x03')
```

Comment:
It would be nice to query the protocol via a more standard API, but as best I can tell it isn't exposed.

Reply:
Right, there isn't even anything obvious in `pickletools` for getting the protocol of a pickle.

Reply:
Actually, I *could* update this to use `pickletools.genops`.

Reply:
Done